### PR TITLE
Updating Contributing Guidelines

### PR DESCRIPTION
**Pull request by ebeahan**
### Added

* Table of contents
* Git/Github guidelines
* Documentation and Additional Information sections.
* Bullet point for ensuring tests pass with `make check`
* Link to CLA

### Changed
* Minor adjustments to existing wording

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -33,22 +53,53 @@ Here are the steps for contributing to ECS.
...
+
+Please follow these guidelines when submitting PRs:
+
+* Include an explanation of your changes in the PR description.
+* Links to relevant issues, external resources, or related PRs are helpful and useful.
+* Update any tests or add new tests where appropriate.
+
+### Issues
+
+Please follow these guidlines when submitting Issues:
```
Comment (webmat) [inline CONTRIBUTING.md:84]:
```suggestion
Please follow these guidelines when submitting Issues:
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -86,6 +137,9 @@ The `type` is the [Elasticsearch field type](https://www.elastic.co/guide/en/ela
...
 `example` adds an sample value.
 
-## Guidelines for implementing ECS
+## Additional Resources
 
-Look at our [Guidelines and Best Practices](https://www.elastic.co/guide/en/ecs/current/ecs-guidelines.html) on the ECS documentation website.
+* [ECS Guidelines and Best Practices](https://www.elastic.co/guide/en/ecs/current/ecs-guidelines.html)
+* [ECS Documentation](https://www.elastic.co/guide/en/ecs/current/index.html)
+* [ECS on Elastic Discuss Forums](https://discuss.elastic.co/tag/elastic-common-schema)
+* [#ecs on the Elasticstack Community Slack](https://elasticstack.slack.com)
```
Comment (webmat) [inline CONTRIBUTING.md:145]: Love this, thanks for adding all of them!

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -33,22 +53,53 @@ Here are the steps for contributing to ECS.
...
+
+To build the docs and open them in your browser:
+
+```bash
+make docs
+```
+
+## Schema Files
 
 The `fields.yml` files describe the Elastic Common Schema in a structured way. We can use these files to generate an Elasticsearch index template, a Kibana index pattern, or documentation output.
```
Comment (webmat) [inline CONTRIBUTING.md:104]: Whoa, I think this is from before my time 😲 I think `fields.yml` actually refers to the files in `schemas/*.yml`

Can you reword this a bit? It no longer needs to talk about possibilities, we do all of this now (except Kibana index patterns, and we won't get into that).

Also, this is a great time to tell reinforce that `schemas/*.yml` is where field additions/modifications are made, but if users are looking to consume ECS to generate something else, they should instead head to `generated/ecs/*.yml`

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -3,28 +3,48 @@
...
   - Run `git checkout -b branch-name`, where `branch-name` is the name you want to give your local branch
 * Do your work.
-  - Make changes to the `.yml` files under the `schemas` directory.
+  - Schema changes will be done in the `.yml` files under the `schemas` directory.
+  - Generator scripts and other tooling resides in the `scripts` directory.
+  - Generated artifacts fall under the `generated` directory.
 * Run `make` to update generated files.
   - Note that the README.md file is generated, and should not be edited directly. Source files are in the /docs directory.
+* If necessary, make sure tests pass.
+  - `make check`
```
Comment (webmat) [inline CONTRIBUTING.md:46]: For running tests, I rather recommend `make test`, which strictly runs the Python tests.

`make check` is an omnibus task that runs the tests, but also checks the code formatting and changes the code in place, and finally it fails if there's anything not checked in (aka if code formatting changed any files). So it's not the best when running the tests while working on a PR

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -3,28 +3,48 @@
...
 
+* Sign the [Contributor License Agreement](http://www.elastic.co/contributor-agreement/).
 * Set up your git environment.
   - Create [your own fork](https://help.github.com/articles/fork-a-repo/) of the ECS repo.
   - Clone your fork to your machine.
 * Create a local branch to hold your changes.
   - Run `git checkout -b branch-name`, where `branch-name` is the name you want to give your local branch
 * Do your work.
-  - Make changes to the `.yml` files under the `schemas` directory.
+  - Schema changes will be done in the `.yml` files under the `schemas` directory.
```
[the remark this answers is not in the document]
Comment (webmat) [reply CONTRIBUTING.md:40]: Actually this is addressed below, so here perhaps just link to the #schema-files section :-)

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -3,28 +3,48 @@
...
+  - Generator scripts and other tooling resides in the `scripts` directory.
+  - Generated artifacts fall under the `generated` directory.
 * Run `make` to update generated files.
   - Note that the README.md file is generated, and should not be edited directly. Source files are in the /docs directory.
```
[the remark this answers is not in the document]
Comment (webmat) [reply CONTRIBUTING.md:44]: > Source files are in the /docs directory.

That's strange wording also. Let's remove the line entirely, and add a mention that "documentation files are in the `docs` directory", to the bullet points above instead.

File: CONTRIBUTING.md (hunk 3 above)
Comment (ebeahan) [reply CONTRIBUTING.md]: Let me know what you think of the updated wording here.

I can see moving the example `agent` snippet into the `schemas/README.md` and linking to it from `CONTRIBUTING.md`. The idea being to have `schemas/README.md` be the single place in the repo with the schema definition, examples, etc.

Comment (webmat) [reply CONTRIBUTING.md]: Yes for now let's completely remove the section about `agent`, I agree.

However please don't modify `schemas/README.md` for now, as I've dramatically improved it in the `nest-as-rampage` branch. You can check it out [here](https://github.com/webmat/ecs/blob/nest-as-rampage/schemas/README.md)

**Pull request merged**